File: appVendedor/app_vendedor_mipedido/lib/components/nav_bar.dart
import 'package:flutter/material.dart';

class NavBar extends StatelessWidget implements PreferredSizeWidget {
  final String title;
  final VoidCallback? onBackPressed;
  final List<Widget>? actions;
  final bool isBackButtonAlert;

  const NavBar({
    super.key,
    required this.title,
    this.onBackPressed,
    this.actions,
    this.isBackButtonAlert = false,
  });

  @override
  Widget build(BuildContext context) {
    final colorScheme = Theme.of(context).colorScheme;

    return AppBar(
      title: Text(
        title,
        style: TextStyle(
          color: colorScheme.onPrimary,
          fontSize: 20,
          fontWeight: FontWeight.bold,
        ),
      ),
      leading:
          onBackPressed != null
              ? IconButton(
                icon: Icon(
                  Icons.arrow_back_sharp,
                  color:
                      isBackButtonAlert
                          ? colorScheme.errorContainer
                          : colorScheme.onPrimary,
                ),
                onPressed: onBackPressed,
              )
              : null,
      actions: actions,
      backgroundColor: colorScheme.primary,
      elevation: 4,
      shadowColor: colorScheme.shadow.withValues(alpha: 0.3),
    );
  }

  @override
  Size get preferredSize => const Size.fromHeight(kToolbarHeight);
}

File: appCliente/app/src/main/java/uk/app02loveslollipop/mipedido/cliente/components/NavBar.kt
package uk.app02loveslollipop.mipedido.cliente.components

import androidx.compose.foundation.layout.WindowInsets
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.ArrowBack
import androidx.compose.material3.*
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * NavBar component that mimics the Flutter app's AppBar style
 */
@OptIn(ExperimentalMaterial3Api::class) //ExperimentalMaterial3Api is used for Material3 components
@Composable
fun NavBar(
    title: String,
    onBackPressed: (() -> Unit)? = null,
    actions: @Composable () -> Unit = {},
    isBackButtonAlert: Boolean = false,
    modifier: Modifier = Modifier
) {
    TopAppBar(
        title = {
            Text(
                text = title,
                style = MaterialTheme.typography.titleLarge.copy(
                    fontSize = 20.sp,
                    fontWeight = FontWeight.Bold
                ),
                color = MaterialTheme.colorScheme.onPrimary
            )
        },
        navigationIcon = {
            if (onBackPressed != null) {
                IconButton(onClick = onBackPressed) {
                    Icon(
                        imageVector = Icons.AutoMirrored.Filled.ArrowBack,
                        contentDescription = "Back",
                        tint = if (isBackButtonAlert) 
                            MaterialTheme.colorScheme.errorContainer
                        else 
                            MaterialTheme.colorScheme.onPrimary
                    )
                }
            }
        },
        actions = { actions() },
        colors = TopAppBarDefaults.topAppBarColors(
            containerColor = MaterialTheme.colorScheme.primary,
            titleContentColor = MaterialTheme.colorScheme.onPrimary,
            navigationIconContentColor = MaterialTheme.colorScheme.onPrimary,
            actionIconContentColor = MaterialTheme.colorScheme.onPrimary
        ),
        modifier = modifier,
        // Adding shadow for elevation effect
        scrollBehavior = TopAppBarDefaults.pinnedScrollBehavior(),
    )
}